lib/cortex/snippets/webpage.rb

```diff
@@ -2,109 +2,123 @@ module Cortex
   module Snippets
     class Webpage
       def initialize(cortex_client, url)
-        @webpage = cortex_client.webpages.get_feed(url).contents
+        @webpage = cortex_client.webpages.get_feed(url)
+        @contents = @webpage.contents
+        @page_cache_key = @webpage.status == 200 ? "view:#{url}@#{@contents.updated_at}" : "view:#{url}"
```

Comment:
This is an application-specific concern. As we can't know the consuming application's conventions for cache key naming (+ it doesn't match ours), I'd suggest this isn't appropriate to offload to a library. Instead, I think this fits better as part of a helper module in our consuming application. We'll definitely need to expose the entire `@webpage` object, however - let's create a getter so that Employer can utilize it.

Reply:
Additionally, I'm not sure we want to change the cache key based on the response. If the response is a failure, let's just update the existing cache object with the failed response rather than swallowing the error here and producing no traceability.

Reply:
If there is an error then the employer page will render without the webpage object, and since this object is cached, the results will remain the same until the cache is busted. Before I make any changes we should have a meeting and discuss the general strategy.

Reply:
I made the changes